packages/menu/src/Menu.tsx

```diff
@@ -207,11 +207,11 @@ function Menu({
           });
         }
 
-        if (props?.children) {
-          return React.cloneElement(child, {
-            children: updateChildren(props.children),
-          });
-        }
+        // if (props?.children) {
+        //   return React.cloneElement(child, {
+        //     children: updateChildren(props.children),
+        //   });
+        // }
```

Comment:
If we no longer need this, can we get rid of this? 
If it may be needed in the future, can we add a note?

Reply:
Dug into this a bit more. Without this block any MenuItems (or SubMenu etc.) will not receive the props they're expecting (e.g. Line 196-7)

Reply:
Do you have a reproduction of these issues? I tried playing around with props and didn't find any issues:

```
<Menu trigger={<button onClick={handleClick}>trigger</button>} open={open}>
          <MenuItem size='large' active>
            Edit Data Source Configuration
          </MenuItem>
          <MenuItem size='default'>
            Test
          </MenuItem>
          <MenuItem size='large'>Edit Rules</MenuItem>
          <Link href="/" passHref>
            <MenuItem disabled size="large">
              Test 2
            </MenuItem>
          </Link>
          <Link href="/" passHref>
            <MenuItem size="large">
              Test 3
            </MenuItem>
          </Link>
          <SubMenu
            title="Submenu"
            description=".design"
            active
            href="http://mongodb.design"
          >
            <MenuItem>SubMenu Item 1</MenuItem>
            <MenuItem>SubMenu Item 2</MenuItem>
            <MenuItem disabled>SubMenu Item 3</MenuItem>
            <Link href="/" passHref>
              <MenuItem size="large">
                SubMenu Item 4
              </MenuItem>
            </Link>
          </SubMenu>
        </Menu>
```

![image](https://user-images.githubusercontent.com/99368875/163033048-f0862b1c-535e-4d66-9bdb-db95a520a920.png)

There's an error with the `<SubMenu>` component above, but that's because it has its own `React.cloneElement` call. Will look into this in the meanwhile, but it'd be good to know what issues you ran into.